Upgrade from rc5-hotfix1 to master-latest failed

<!-- Please state your issue using the following template and, most importantly, in English. -->

#### Steps/Code to reproduce:
* deploy milvus `v2.0.0-rc5-hotfix1` by `docker-compose up -d`
* insert data
* uninstall milvus by `docker-compose down` but keep volumes dir
* deploy milvus `master-20210908-50c677f` by `docker-compose up -d`

#### Expected result:
deploy sucess

#### Actual results:
containers like`milvus-standalone` exited 
```
[2021/09/08 11:16:11.422 +00:00] [DEBUG] [dml_channels.go:55] ["init dml channels"] [num=256]
[2021/09/08 11:16:11.422 +00:00] [ERROR] [dml_channels.go:101] ["invalid channel name"] [chanName=IVF_FLAT_427582950415007745_0] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).AddProducerChannels\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:101\ngithub.com/milvus-io/milvus/internal/rootcoord.(*Core).Init.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:932\nsync.(*Once).doSlow\n\t/usr/local/go/src/sync/once.go:66\nsync.(*Once).Do\n\t/usr/local/go/src/sync/once.go:57\ngithub.com/milvus-io/milvus/internal/rootcoord.(*Core).Init\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:860\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:182\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:107\ngithub.com/milvus-io/milvus/cmd/components.(*RootCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/root_coord.go:45\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runRootCoord.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:99"]
panic: invalid channel name

goroutine 74 [running]:
github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).AddProducerChannels(0xc000650f80, 0xc0000ad800, 0x16, 0x20)
        /go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:102 +0x605
github.com/milvus-io/milvus/internal/rootcoord.(*Core).Init.func1()
        /go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:932 +0x62c
sync.(*Once).doSlow(0xc00015e100, 0xc000669ce8)
        /usr/local/go/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:57
github.com/milvus-io/milvus/internal/rootcoord.(*Core).Init(0xc00015e000, 0x273fc20, 0xc00023fd10)
        /go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:860 +0x41e
github.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).init(0xc0002b2140, 0x100000000, 0xc0004fa5c8)
        /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:182 +0x336
github.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).Run(0xc0002b2140, 0xffffffffffffffff, 0x2728740)
        /go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:107 +0x2f
github.com/milvus-io/milvus/cmd/components.(*RootCoord).Run(...)
        /go/src/github.com/milvus-io/milvus/cmd/components/root_coord.go:45
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runRootCoord.func1(0x2597001, 0x2727e40, 0xc0000b5480, 0xc00041d590, 0xc0004fa5c0)
        /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:99 +0xdc
created by github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runRootCoord
        /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:84 +0xb0
```

#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93): rc5-hotfix1 --> 50c677f
- Deployment mode(standalone or cluster): both standalone and cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus v2.0.0rc5
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: 
- GPU: 
- Others: 

#### Configuration file:


#### Additional context:

[cluster-upgrade-2021-09-08-10-08-15.log](https://github.com/milvus-io/milvus/files/7128282/cluster-upgrade-2021-09-08-10-08-15.log)
[standalone-upgrade-2021-09-08-10-07-33.log](https://github.com/milvus-io/milvus/files/7128284/standalone-upgrade-2021-09-08-10-07-33.log)

--- a/internal/rootcoord/dml_channels.go
+++ b/internal/rootcoord/dml_channels.go
@@ -29,6 +29,7 @@ type dmlChannels struct {
 	refcnt     sync.Map
 	idx        *atomic.Int64
 	pool       sync.Map
+	dml        sync.Map
 }
 
 func newDmlChannels(c *Core, chanNamePrefix string, chanNum int64) *dmlChannels {
@@ -39,6 +40,7 @@ func newDmlChannels(c *Core, chanNamePrefix string, chanNum int64) *dmlChannels 
 		refcnt:     sync.Map{},
 		idx:        atomic.NewInt64(0),
 		pool:       sync.Map{},
+		dml:        sync.Map{},
 	}
 
 	var i int64
@@ -83,12 +85,18 @@ func (d *dmlChannels) GetNumChannels() int {
 func (d *dmlChannels) Broadcast(chanNames []string, pack *msgstream.MsgPack) error {
 	for _, chanName := range chanNames {
 		// only in-use chanName exist in refcnt
-		if _, ok := d.refcnt.Load(chanName); !ok {
-			return fmt.Errorf("channel %s not exist", chanName)
-		}
-		v, _ := d.pool.Load(chanName)
-		if err := (*(v.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
-			return err
+		if _, ok := d.refcnt.Load(chanName); ok {
+			v, _ := d.pool.Load(chanName)
+			if err := (*(v.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
+				return err
+			}
+		} else {
+			log.Debug("broadcast to old version channel", zap.String("chanName", chanName))
+			if ds, ok := d.dml.Load(chanName); ok {
+				if err := (*(ds.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
+					return err
+				}
+			}
 		}
 	}
 	return nil
@@ -97,10 +105,7 @@ func (d *dmlChannels) Broadcast(chanNames []string, pack *msgstream.MsgPack) err
 // AddProducerChannels add named channels as producer
 func (d *dmlChannels) AddProducerChannels(names ...string) {
 	for _, name := range names {
-		if _, ok := d.pool.Load(name); !ok {
-			log.Error("invalid channel name", zap.String("chanName", name))
-			panic("invalid channel name")
-		} else {
+		if _, ok := d.pool.Load(name); ok {
 			var cnt int64
 			if _, ok := d.refcnt.Load(name); !ok {
 				cnt = 1
@@ -110,6 +115,15 @@ func (d *dmlChannels) AddProducerChannels(names ...string) {
 			}
 			d.refcnt.Store(name, cnt)
 			log.Debug("assign dml channel", zap.String("chanName", name), zap.Int64("refcnt", cnt))
+		} else {
+			log.Debug("add old version channel name", zap.String("chanName", name))
+			ms, err := d.core.msFactory.NewMsgStream(d.core.ctx)
+			if err != nil {
+				log.Error("add msgstream failed", zap.String("name", name), zap.Error(err))
+				panic("add msgstream failed")
+			}
+			ms.AsProducer([]string{name})
+			d.dml.Store(name, &ms)
 		}
 	}
 }
@@ -117,13 +131,18 @@ func (d *dmlChannels) AddProducerChannels(names ...string) {
 // RemoveProducerChannels removes specified channels
 func (d *dmlChannels) RemoveProducerChannels(names ...string) {
 	for _, name := range names {
-		v, ok := d.refcnt.Load(name)
-		if ok {
+		if v, ok := d.refcnt.Load(name); ok {
 			cnt := v.(int64)
 			if cnt > 1 {
 				d.refcnt.Store(name, cnt-1)
 			} else {
 				d.refcnt.Delete(name)
+			}
+		} else {
+			log.Debug("remove old version channel name", zap.String("chanName", name))
+			if ds, ok := d.dml.Load(name); ok {
+				(*(ds.(*msgstream.MsgStream))).Close()
+				d.dml.Delete(name)
 			}
 		}
 	}

--- a/internal/rootcoord/dml_channels_test.go
+++ b/internal/rootcoord/dml_channels_test.go
@@ -13,7 +13,6 @@ package rootcoord
 
 import (
 	"context"
-	"fmt"
 	"testing"
 
 	"github.com/milvus-io/milvus/internal/msgstream"
@@ -46,12 +45,20 @@ func TestDmlChannels(t *testing.T) {
 	chanNames := dml.ListChannels()
 	assert.Equal(t, 0, len(chanNames))
 
+	//randStr := funcutil.RandomString(8)
+	//assert.Panics(t, func() { dml.AddProducerChannels(randStr) })
+	//
+	//err = dml.Broadcast([]string{randStr}, nil)
+	//assert.NotNil(t, err)
+	//assert.EqualError(t, err, fmt.Sprintf("channel %s not exist", randStr))
+
 	randStr := funcutil.RandomString(8)
-	assert.Panics(t, func() { dml.AddProducerChannels(randStr) })
+	dml.AddProducerChannels(randStr)
 
 	err = dml.Broadcast([]string{randStr}, nil)
-	assert.NotNil(t, err)
-	assert.EqualError(t, err, fmt.Sprintf("channel %s not exist", randStr))
+	assert.Nil(t, err)
+
+	dml.RemoveProducerChannels(randStr)
 
 	// dml_xxx_0 => {chanName0, chanName2}
 	// dml_xxx_1 => {chanName1}

